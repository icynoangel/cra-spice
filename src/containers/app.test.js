import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
import { shallow, mount } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

describe('<App />', function () {

  beforeEach(() => {
    this.dictionary = {
      locale: 'en-EN',
      messages: {},
      fetched: true
    };
    this.fetchDictionary = jest.fn();

    this.getComponent = () => {
      return (
        <App.WrappedComponent
          dictionary={this.dictionary}
          fetchDictionary={this.fetchDictionary}
        />
      );
    };
  });

  it('should render <App />', () => {
    const wrapper = shallow(this.getComponent());
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('should fetch dictionary when component mounts', () => {
    const wrapper = shallow(this.getComponent());
    expect(this.fetchDictionary).toHaveBeenCalled();
  });
});
