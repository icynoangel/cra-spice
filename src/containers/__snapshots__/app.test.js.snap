// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render <App /> 1`] = `
<IntlProvider
  key="en-EN"
  locale="en-EN"
  messages={Object {}}
>
  <div
    className="app"
  >
    <div
      className="app-content"
    >
      <InjectIntl(ExampleClassComponent)
        initialCounter={10}
      />
      <InjectIntl(ExampleFunctionComponent)
        value={100}
      />
    </div>
  </div>
</IntlProvider>
`;
